redfish-core/include/utils/name_utils.hpp

```diff
@@ -0,0 +1,79 @@
+
+#pragma once
+#include <async_resp.hpp>
+
+#include <algorithm>
+#include <string>
+#include <variant>
+#include <vector>
+
+namespace redfish
+{
+namespace name_util
+{
+
+/**
+ * @brief Populate the collection "Members" from a GetSubTreePaths search of
+ *        inventory
+ *
+ * @param[i,o] asyncResp  Async response object
+ * @param[i]   path       D-bus object path to find the find pretty name
+ * @param[i]   services   List of services to exporting the D-bus object path
+ * @param[i]   namePath   Json pointer to the name field to update.
+ *
+ * @return void
+ */
+inline void getPrettyName(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& path,
+    const std::vector<std::pair<std::string, std::vector<std::string>>>&
+        services,
```

Comment:
Unlike 1060, this `getPrettyName()` seems using the array of `ServiceMap`, and this function checks the number of services and generates an error here if it is not 1.

The error (eg. internalError()) is better to be generated by the caller based on its context.
So, I think 1060 logic is better.

Caller is doing like
```
            name_util::getPrettyName(asyncResp, path, connectionNames[0].first,
                                     "/Name"_json_pointer);
```
and this getPrettyName is doing like
```
inline void getPrettyName(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                          const std::string& path, const std::string& service,
                          const nlohmann::json::json_pointer& namePath)
```

Reply:
Can we add the part of https://github.com/ibm-openbmc/bmcweb/pull/698 to do this?
(Not other parts).